rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -1467,6 +1507,11 @@ rmw_create_client(
   free_rmw_client.cancel();
   free_service_name.cancel();
 
+  if (TRACETOOLS_TRACEPOINT_ENABLED(rmw_client_init)) {
+    auto client_data = static_cast<rmw_zenoh_cpp::ClientData *>(rmw_client->data);
```

Comment:
We don't really need to re-cast this from the `rmw_client->data` pointer; we can actually much more nicely use the shared_ptr.  What I'd suggest here is that above, before we do the `static_cast<void *>` to put this into `rmw_client->data`, we actually do:

```
ClientDataPtr client_data = node_data->get_client_data(rmw_client);
```

Then we can do `rmw_client->data = static_cast<void *>(client_data.get());` in line 1495, and use the `client_data->copy_gid()` in the next line below.

Reply:
changed!